frontend/src/pages/signin-page.tsx

```diff
@@ -0,0 +1,105 @@
+import { Link, useNavigate } from 'react-router-dom';
+import AddGoogleIcon from '@/assets/svg/google-color-icon.svg';
+import AddGithubIcon from '@/assets/svg/github-icon.svg';
+import { useForm } from 'react-hook-form';
+import type { FieldValues } from 'react-hook-form';
+import { TSignInSchema, signInSchema } from '@/lib/types';
+import 'react-toastify/dist/ReactToastify.css';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { toast } from 'react-toastify';
+
+function signin() {
+  const navigate = useNavigate();
+
+  const { register, 
+        handleSubmit, 
+        formState: { errors , isSubmitting},
+        reset
+      } = useForm<TSignInSchema>({resolver: zodResolver(signInSchema)});
+
+  const onSubmit = async (data: FieldValues) => {
+
+    if(data.email === "abc@gamil.com"){
+      toast.error('Submitting form is failed');
+      return;
+    }
+    
+    // TODO: Server-side validation
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+    
+    reset();
+    navigate("/");
+  }
+
+  return (
+    <div className="flex-grow cursor-default bg-white px-5 py-9 mt-1">
+      <div className="mb-4 flex justify-center">
+        <div className="flex items-center justify-center space-x-4 w-5/6">
+          <h2 className="text-center text-[22px] sm:text-3xl font-bold text-black w-3/4">
+            Sign in to WanderLust
```

Comment:
22px, no hardcoding please, use the relevant tailwind-css, if it doesn't have that... create a custom size variable in the tailwind.config and then use it ....

Reply:
Okay.